gateway/ln-gateway/src/lnd.rs

```diff
@@ -394,31 +393,46 @@ impl ILnRpcClient for GatewayLndClient {
             // LND API allows fee limits in the `i64` range, but we use `u64` for
             // max_fee_msat. This means we can only set an enforceable fee limit
             // between 0 and i64::MAX
-            let fee_cap: i64 = max_fee_msat
+            let fee_limit_msat: i64 = max_fee_msat
                 .try_into()
                 .map_err(|_| anyhow::anyhow!("max_fee_msat exceeds valid LND fee limit ranges"))?;
 
-            let send_response = client
-                .lightning()
-                .send_payment_sync(SendRequest {
-                    payment_request: invoice.to_string(),
-                    fee_limit: Some(FeeLimit {
-                        limit: Some(Limit::FixedMsat(fee_cap)),
-                    }),
+            let payments = client
+                .router()
+                .send_payment_v2(SendPaymentRequest {
+                    payment_request: invoice,
+                    allow_self_payment: true,
+                    no_inflight_updates: true,
+                    timeout_seconds: LND_PAYMENT_TIMEOUT_SECONDS,
+                    fee_limit_msat,
                     ..Default::default()
                 })
                 .await
-                .map_err(|e| anyhow::anyhow!(format!("LND error: {e:?}")))?
-                .into_inner();
-
-            if send_response.payment_preimage.is_empty() {
-                return Err(GatewayError::LnRpcError(tonic::Status::new(
-                    tonic::Code::Internal,
-                    "LND did not return a preimage",
-                )));
-            };
+                .map_err(|e| {
+                    GatewayError::Other(anyhow!("Failed to make outgoing payment: {e:?}"))
+                })?;
 
-            send_response.payment_preimage
+            match payments
+                .into_inner()
+                .message()
+                .await
+                .map_err(|_| GatewayError::ClientNgError)?
```

Comment:
```suggestion
                .map_err(|e| GatewayError::Other(anyhow!("Failed to get payment status: {e:?}")))?
```

Reply:
I think here you should loop getting .message() until `.status` is `SUCCEEDED`

Reply:
> I think here you should loop getting .message() until .status is SUCCEEDED or you get none

As mentioned offline, I don't think that is necessary since we're using the `no_inflight_updates` flag. We should only receive one update - if the payment succeeds or fails.